Main.elm

```diff
@@ -73,30 +75,69 @@ setupControlsView address player =
 shipFieldView : Signal.Address Action -> Ship.Ship -> Html.Html
 shipFieldView address ship =
   if not ship.added then
-    Html.div []
+    Html.div [ Html.Attributes.style [("margin-bottom", "5px")] ]
     [ Html.input -- Ship Row Input
       [ Html.Attributes.value (toString (Ship.getRow ship))
       , Html.Events.on "input" Html.Events.targetValue (Signal.message address << SetupRowField ship.id)
       ]
       []
     , Html.input -- Ship Column Input
-      [ Html.Attributes.value (toString (Ship.getColumn ship))
+      [ Html.Attributes.style [("margin-right", "5px")]
+      , Html.Attributes.value (toString (Ship.getColumn ship))
       , Html.Events.on "input" Html.Events.targetValue (Signal.message address << SetupColumnField ship.id)
       ]
       []
-    , Html.input -- Ship Orientation
-      [ Html.Attributes.type' "radio"
-      , Html.Attributes.checked (ship.orientation == Ship.Horizontal)
-      , Html.Events.onClick address (SetupOrientationField ship.id)
-      ]
-      []
+      -- Ship orientation:
+    , orientationSwitch address ship.id ship.orientation
     , Html.button -- Add Ship
       [ Html.Events.onClick address (SetupAddShip ship.id)
       ]
-      []
+      [ Html.text "Add ship" ]
     ]
   else Html.div [] []
 
+orientationSwitch : Signal.Address Action -> Int -> Ship.Orientation -> Html.Html
+orientationSwitch address shipId orientation =
```

Comment:
The opponent could send their board to the server and then the server sends the tracking board to the client, removing the information pertaining to the opponents ship positions. But this is not related to ui at all. This is more of a design decision.